src/vs/workbench/common/contextkeys.ts

```diff
@@ -77,6 +77,7 @@ export const IsCenteredLayoutContext = new RawContextKey<boolean>('isCenteredLay
 export const SplitEditorsVertically = new RawContextKey<boolean>('splitEditorsVertically', false, localize('splitEditorsVertically', "Whether editors split vertically"));
 export const EditorAreaVisibleContext = new RawContextKey<boolean>('editorAreaVisible', true, localize('editorAreaVisible', "Whether the editor area is visible"));
 export const EditorTabsVisibleContext = new RawContextKey<boolean>('editorTabsVisible', true, localize('editorTabsVisible', "Whether editor tabs are visible"));
+export const EditorPinnedAndUnpinnedTabsContext = new RawContextKey<boolean>('editorPinnedAndUnpinnedTabsVisible', false, localize('editorPinnedAndUnpinnedTabsVisible', "Whether at least one pinned and one unpinned editor tab is visible"));
```

Comment:
I am thinking to simplify this: we could just track if an editor is pinned in a group to show the action. Because that shows intent to use pinned tabs. Maybe we make this `ActiveEditorGroupHasPinnedTabs` and set this conditionally per group?

Reply:
Even though it might not happen often, I don't like the case where all tabs are pinned. We are using the context menu so that users can better discover this feature, and if they toggle the setting and nothing happens, they might just toggle it back as they don't see anything immediately change...

Reply:
Hide it from users